vector/src/main/java/im/vector/app/features/usercode/UserCodeActivity.kt

```diff
@@ -106,23 +114,31 @@ class UserCodeActivity : VectorBaseActivity<ActivitySimpleBinding>(),
                 }
             }
         }
+
+        qrViewModel.observeViewEvents {
+            when (it) {
+                is QrCodeScannerEvents.CodeParsed  -> {
+                    sharedViewModel.handle(UserCodeActions.DecodedQRCode(it.result))
+                }
+                QrCodeScannerEvents.SwitchMode     -> {
+                    sharedViewModel.handle(UserCodeActions.SwitchMode(UserCodeState.Mode.SHOW))
+                }
+                is QrCodeScannerEvents.ParseFailed -> {
+                    Toast.makeText(this, R.string.qr_code_not_scanned, Toast.LENGTH_SHORT).show()
+                    finish()
+                }
+            }
+        }
     }
 
     override fun onDestroy() {
         supportFragmentManager.unregisterFragmentLifecycleCallbacks(fragmentLifecycleCallbacks)
         super.onDestroy()
     }
 
-    private fun showFragment(fragmentClass: KClass<out Fragment>, bundle: Bundle) {
+    private fun showFragment(fragmentClass: KClass<out Fragment>, params: Parcelable? = null) {
         if (supportFragmentManager.findFragmentByTag(fragmentClass.simpleName) == null) {
```

Comment:
I don't remember how `replace` will work in case if there is a fragment with a same tag, so not sure if this check is really needed. If not, we could remove entire method. @bmarty maybe you could tell if it is obsolete or not?

Reply:
We can keep it like that for now.